core/staking_verifier.go

```diff
@@ -313,11 +313,6 @@ func VerifyAndCollectRewardsFromDelegation(
 				Msg("Delegation index out of bound")
 			return nil, nil, errors.New("Delegation index out of bound")
 		}
-		if err := wrapper.SanityCheck(
```

Comment:
why sanity check not required?

Reply:
it's redundant, because CollectReward doesn't affect any part of the sanity checks. Meaning it will never fail the check.